Bottom: 3d855072e72bf35554a30699711a96a7711bc8c0
Top:    c17e26d6fc5e5d0430915fd5d9eca2c986f96291
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-17 20:08:56 +0000

Revert "Temporarily disable the circular contour radius adjustment"

This reverts commit 9e74e901dd8fdce88e566aa2757cbc722d22939d.

Was part of adding arc support, now temporarily borrowed for debugging


---

diff --git a/src/polygon.h b/src/polygon.h
index 369efea..a02fd99 100644
--- a/src/polygon.h
+++ b/src/polygon.h
@@ -37,28 +37,11 @@
 #define POLY_CIRC_SEGS 40
 #define POLY_CIRC_SEGS_F ((float)POLY_CIRC_SEGS)
 
-#if 0
-/* THIS IS BROKEN:
- *
- * IT BREAKS THE CIRCULARITY OF CIRULAR CONTORS, AS THE FIRST
- * FIRST VERTEX ADDED BY CirclePoly IS NOT RADIUS ADJUSTED.
- *
- * IT BREAKS CIRCULARITY OF ALIGMENT BETWEEN A LINE AND ITS END-CAPS,
- * LEADING TO MORE COMPLEX CONTOURS FOR COMMON LINE-LINE INTERSECTIONS,
- * SUCH AS 90 AND 45 DEGREE ANGLES
- *
- * IT WAS INTENDED TO AVOID DRC ERRORS WITH "TOO-CLOSE" FEATURES,
- * BUT COULD OTHERWISE CAUSE THEM FOR "TOO THIN" FEATURES - INSIDE/OUTSIDE
- * CONTOUR APPROXIMATION NEEDS TO BE CONTROLED DEPENDING ON THE REQUIREMENT
- */
 /* adjustment to make the segments outline the circle rather than connect
  * points on the circle: 1 - cos (\alpha / 2) < (\alpha / 2) ^ 2 / 2
  */
 #define POLY_CIRC_RADIUS_ADJ (1.0 + M_PI / POLY_CIRC_SEGS_F * \
                                     M_PI / POLY_CIRC_SEGS_F / 2.0)
-#else
-#define POLY_CIRC_RADIUS_ADJ 1.0
-#endif
 
 /* polygon diverges from modelled arc no more than MAX_ARC_DEVIATION * thick */
 #define POLY_ARC_MAX_DEVIATION 0.02
